domain/src/main/kotlin/com/studentcenter/weave/domain/meeting/event/MeetingCompletedEvent.kt

```diff
@@ -0,0 +1,16 @@
+package com.studentcenter.weave.domain.meeting.event
+
+import com.studentcenter.weave.domain.common.DomainEvent
+import com.studentcenter.weave.domain.meeting.entity.Meeting
+
+data class MeetingCompletedEvent(
+    override val entity: Meeting,
+) : DomainEvent<Meeting> {
+
+    companion object {
+        fun Meeting.createCompletedEvent(): MeetingCompletedEvent {
+            return MeetingCompletedEvent(this)
+        }
```

Comment:
p2) 컨디션 체크 하면 좋을 것 같아요!
```suggestion
        fun Meeting.createCompletedEvent(): MeetingCompletedEvent {
        		if (this.isCompleted().not()) {
        		  // loggin? 
        		  return
        		}
            return MeetingCompletedEvent(this)
        }
```

Reply:
require로 예외발생시키도록 하는건 어떤가요?

Reply:
예외를 던져서 이벤트 발행이 안된다면 롤백시킬 의도라면 에러도 좋아요!
안전하게 한다면 이벤트 발행을 DB에 기록하고 커밋한 뒤에 이벤트 발행 시도를 하고 실패해도 에러가  아닌 로깅을 하고 추후에 리트라이 하는 방식도 괜찮다고 생각했는데 요건 조금 복잡한 느낌이네요!